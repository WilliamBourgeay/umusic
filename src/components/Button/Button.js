// @flow

import React from 'react';
import Button from '@material-ui/core/Button';

type Props = {
  variant: string,
  text: string,
  color: string,
}

const SubmitButton = ({ variant, text, color }: Props) => (
  <Button
    variant={variant}
    color={color}
  >
    {text}
  </Button>
);

export default SubmitButton;
