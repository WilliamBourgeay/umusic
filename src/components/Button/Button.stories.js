import React from 'react';
import { storiesOf } from '@storybook/react';
import SubmitButton from './Button';

const stories = storiesOf('Button', module);

stories
  .add('SubmitButton', () => (
    <SubmitButton text="Accept" variant="contained" color="primary" />
  ));
