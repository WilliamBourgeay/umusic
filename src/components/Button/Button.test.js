import * as React from 'react';
import renderer from 'react-test-renderer';

import Button from './Button';

describe('Button', () => {
  it('should render a Button', () => {
    const tree = renderer.create(<Button variant="contained" text="hehe" color="primary" />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
