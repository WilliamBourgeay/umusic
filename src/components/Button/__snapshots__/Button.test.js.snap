// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a Button 1`] = `
<button
  className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-containedPrimary-13 MuiButton-raised-15 MuiButton-raisedPrimary-16"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-2"
  >
    hehe
  </span>
  <span
    className="MuiTouchRipple-root-30"
  />
</button>
`;
